pkg/daemon/bootc.go

```diff
@@ -0,0 +1,245 @@
+	c := client.newCmd(args...)
+	return c.Run()
+}
+
+// QueryStatus loads the current system state.
+func (client *BootcClient) QueryStatus() (*Status, error) {
+	var q Status
+	c := client.newCmd("status", "--json")
+	buf, err := c.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(buf, &q); err != nil {
+		return nil, fmt.Errorf("failed to parse `bootc status --json` output: %w", err)
+	}
+
+	return &q, nil
+}
+
+// GetBootedImage finds the booted image, or returns nil if none is found.
+func (s *Status) GetBootedImage() *BootEntry {
+	return s.Booted
+}
+
+// GetStagedImage finds the staged image for the next boot, or returns nil if none is found.
+func (s *Status) GetStagedImage() *BootEntry {
+	return s.Staged
+}
+
+// GetRollbackImage finds the rollback deployment, or returns nil if none is found.
+func (s *Status) GetRollbackImage() *BootEntry {
+	return s.Rollback
+}
+
+// GetBaseChecksum returns the ostree commit used as a base.
+func (b *BootEntry) GetBaseChecksum() string {
+	return b.Ostree.Checksum
+}
+
+// GetContainerImageReference returns the the container image reference.
+func (b *BootEntry) GetContainerImageReference() string {
+	return b.Image.Image.Image
+}
+
+// imgref.go
+// ImageReference captures an image signature verification policy alongside an image reference.
+// This mirrors https://github.com/containers/bootc/blob/94521ecb19b8d0f2bfc36220322c60cafa036296/lib/src/spec.rs#L69
+type ImageReference struct {
+	Image     string          `json:"image"`
+	Transport string          `json:"transport"`
+	Signature *ImageSignature `json:"signature,omitempty"`
+}
+
+// ImageSignature mirrors https://github.com/containers/bootc/blob/94521ecb19b8d0f2bfc36220322c60cafa036296/lib/src/spec.rs#L57
+// Same as https://docs.rs/ostree-ext/latest/ostree_ext/container/enum.SignatureSource.html
+type ImageSignature struct {
+	AllowInsecure bool
+	OstreeRemote  string
+}
+
+// bootc.go
+// Synchronously invoke bootc, writing its stdout to our stdout,
+// and gathering stderr into a buffer which will be returned in err
+// in case of error.
+func runBootc(args ...string) error {
+	return runCmdSync("bootc", args...)
+}
+
+func (b *BootcClient) Initialize() error {
+	// Commands like update and rebase need the pull secrets to pull images and manifests,
+	// make sure we get access to them when we Initialize
+	err := useMergedPullSecrets(bootcSystem)
+	if err != nil {
+		return fmt.Errorf("Error while ensuring access to kublet config.json pull secrets: %w", err)
+	}
+
+	return nil
+}
+
+// GetBootedAndStagedImage returns the current booted and staged image found
+func (b *BootcClient) GetBootedAndStagedImage() (booted, staged *BootEntry, err error) {
+	status, err := b.QueryStatus()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	booted = status.GetBootedImage()
+	staged = status.GetStagedImage()
+
+	return
+}
+
+// // GetStatus returns multi-line human-readable text describing system status
```

Comment:
I am having this block here but commented out because according to https://github.com/containers/bootc/issues/408, it seems that the bootc team is actively working on creating options for outputting human-readable bootc status, which is not ready yet. By the time we merge this PR, if they have something handy for us to use, I think we can un-comment the block and bring it in; while if not, then we should delete it :)

Reply:
That makes sense :smile: . We can leave it in that case.